api/api.go

```diff
@@ -1,18 +1,22 @@
 package api
 
-import "github.com/mesg-foundation/core/container"
+import (
+	"github.com/mesg-foundation/core/container"
+	"github.com/mesg-foundation/core/database"
+)
 
 // API exposes all functionalies of MESG core.
 type API struct {
+	db        *database.ServiceDB
 	container *container.Container
 }
 
 // Option is a configuration func for MESG.
 type Option func(*API)
 
 // New creates a new API with given options.
-func New(options ...Option) (*API, error) {
-	a := &API{}
+func New(db *database.ServiceDB, options ...Option) (*API, error) {
```

Comment:
lets pass _db_ as an _Option_ instead. and check if db is not nil and return an error if needed after options are applied.

Reply:
I don't want to, because it is required option and not an optional one. here you have validation in compile time and not runtime.